DeployJenkinsfile

```diff
@@ -20,156 +21,252 @@ pipeline {
+                                reportDir: 'uaa/build/reports/jacoco/jacocoRootReportAcceptanceTest/html',
+                                reportFiles: 'index.html',
+                                reportName: 'Acceptance Test Code Coverage'
+                            ]
+                        }
+                    }
+                }
+                stage('Upload UAA Artifact upon successful deploy of staging artifact') 
+                {
+                    agent {
+                        label 'dind'
+                    }
+                    when {
+                        expression { params.DEPLOYMENT_TYPE == 'cf3-staging' }
+                    }
+                    steps{
+                        dir('deploy-workspace') {
+                            unstash 'uaa-war'
+                        }
+
+                        dir('uaa') {
+                            checkout scm
+                        }
+
+                        dir('uaa/iam-k8s-utils') {
+                            git changelog: false, credentialsId: 'github.build.ge.com', poll: false,
+                                url: 'https://github.build.ge.com/predix/iam-k8s-utils.git'
+                        }
+
+                        script {
+                            def util = load('uaa/JenkinsfileCommon.groovy')
+                            APP_VERSION = util.getAppVersion()
+                            echo "Publishing UAA ${APP_VERSION} Artifact to Artifactory"
+                            def uploadSpec = """{
+                                "files": [
+                                    {
+                                        "pattern": "deploy-workspace/cloudfoundry-identity-uaa-${APP_VERSION}.war",
+                                        "target": "MAAXA/org/cloudfoundry/identity/cloudfoundry-identity-uaa/${APP_VERSION}/"
+                                    }
+                                ]
+                            }"""
+                            def buildInfo = buildGeArtServer.upload(uploadSpec)
+                            buildGeArtServer.publishBuildInfo(buildInfo)
+                        }
+                    }
                 }
             }
         }
-        stage('Upload UAA Artifact upon successful deploy of staging artifact') {
-            agent {
-                label 'dind'
+        stage('EKS Deploy')
+        {
+            when 
+            {
+                expression { params.DEPLOY_EKS == true }
             }
-            when {
-                expression { params.DEPLOYMENT_TYPE == 'cf3-staging' }
-            }
-            steps{
-                dir('deploy-workspace') {
-                    unstash 'uaa-war'
-                }
-
-                dir('uaa') {
-                    checkout scm
+            agent{
+                docker {
+                    image "${NODE['IMAGE']}"
+                    label "${NODE['LABEL']}"
+                    args "${NODE['ARGS']}"
```

Comment:
Minor, but for EKS Deploy stage we could use simple agent same as for "Upload UAA Artifact..." stage:
```
agent {
    label 'dind'
}
```